File: app/src/main/java/com/wolkowiczmateusz/android/cleanmvpretrofitrxdaggerboilerplate/presentation/customviews/progressDialog/DialogTypeStrategy.java
package com.wolkowiczmateusz.android.cleanmvpretrofitrxdaggerboilerplate.presentation.customviews.progressDialog;

import android.content.Context;

public class DialogTypeStrategy {

    Dialog dialog;

    public DialogTypeStrategy(Dialog dialog) {
        this.dialog = dialog;
    }

    public void show(Context context, String msg) {
        dialog.showDialog(context, msg);
    }

    public void dismiss() {
        dialog.dismiss();
    }
}

File: app/src/main/java/com/wolkowiczmateusz/android/cleanmvpretrofitrxdaggerboilerplate/presentation/customviews/progressDialog/DialogTypeStrategy.kt
package com.wolkowiczmateusz.android.cleanmvpretrofitrxdaggerboilerplate.presentation.customviews.progressDialog

import android.content.Context

class DialogTypeStrategy(var dialog: Dialog) {

    fun show(context: Context, msg: String) {
        dialog.showDialog(context, msg)
    }

    fun dismiss() {
        dialog.dismiss()
    }
}
